File: api/src/buildings/dto/CreateBuildingDto.py
from pydantic import BaseModel, Field
from typing import Optional, List

class CreateBuildingDto(BaseModel):
    name: str = Field(..., min_length=1)
    university: str = Field(..., min_length=1)
    description: Optional[str] = None
    imageUrls: List[str] = Field(default_factory=list)
    address: str = Field(..., min_length=1)
    latitude: float = Field(..., ge=-90, le=90)
    longitude: float = Field(..., ge=-180, le=180)

File: mobile/app/src/main/java/com/example/booking/data/remote/model/CreateBuildingDto.kt
package com.example.booking.data.remote.model

import android.net.Uri

data class CreateBuildingDto(
    val name: String,
    val university: String,
    val description: String? = null,
    val address: String,
    val latitude: Float,
    val longitude: Float,
    val images: List<Uri> = emptyList()
)